src/stm32/gpioperiph.c

```diff
@@ -33,6 +34,13 @@ gpio_peripheral(uint32_t gpio, uint32_t mode, int pullup)
     // stm32f4 is ~50Mhz at 40pF
     // stm32g0 is ~30Mhz at 50pF
     // stm32h7 is ~85Mhz at 50pF
-    uint32_t ospeed = CONFIG_MACH_STM32F0 ? 0x01 : 0x02;
     regs->OSPEEDR = (regs->OSPEEDR & ~m_msk) | (ospeed << m_shift);
 }
+
+// Set the mode and extended function of a pin
+void
+gpio_peripheral(uint32_t gpio, uint32_t mode, int pullup)
+{
+    uint32_t ospeed = CONFIG_MACH_STM32F0 ? 0x01 : 0x02;
+    gpio_peripheral_speed(gpio, mode, pullup, ospeed);
+}
```

Comment:
If it is necessary to configure the ospeed parameter, then add a new `GPIO_HIGHSPEED` flag to the existing `mode` parameter.  Also, please don't claim a copyright on gpio.c for such a minor change.

Reply:
Proposed changes included in latest commit. Removed the copyright claim. Sorry for introducing it for such a small change. I was blind there when I added the copyright claims to my changed / introduced files.